src/ts/paging/paging.ts

```diff
@@ -1,84 +1,90 @@
-  if (selectedPageIndex === pageIndex) {
-    return;
+  /**
+   * Returns selected pages
+   * @returns WaterfallData - currently selected page
+   */
+  public getSelectedPage(): WaterfallData {
+    return this.doc.pages[this.selectedPageIndex];
   }
-  if (pageIndex < 0 || pageIndex >=  getPageCount()) {
-    throw new Error("Page does not exist - Invalid pageIndex selected");
+
+  /**
+   * Returns index of currently selected page
+   * @returns number - index of current page (0 based)
+   */
+  public getSelectedPageIndex(): number {
+    return this.selectedPageIndex;
   }
 
-  selectedPageIndex = pageIndex;
-  let selectedPage = waterfallDocsService.getDocs().pages[selectedPageIndex];
-  onPageUpdateCbs.forEach((cd) => {
-    cd(selectedPageIndex, selectedPage);
-  });
-}
+  /**
+   * Update which pageIndex is currently update.
+   * Published `onPageUpdate`
+   * @param  {number} pageIndex
+   */
+  public setSelectedPageIndex(pageIndex: number) {
+    if (this.selectedPageIndex === pageIndex) {
+      return;
+    }
+    if (pageIndex < 0 || pageIndex >=  this.getPageCount()) {
+      throw new Error("Page does not exist - Invalid pageIndex selected");
+    }
 
-/**
- * Register subscriber callbacks to be called when the pageindex updates
- * @param  {OnPagingCb} cb
- * @returns number - index of the callback
- */
-export function onPageUpdate(cb: OnPagingCb): number {
-  if (getPageCount() > 1) {
-    return onPageUpdateCbs.push(cb);
+    this.selectedPageIndex = pageIndex;
+    let selectedPage = this.doc.pages[this.selectedPageIndex];
+    this.onPageUpdateCbs.forEach((cd) => {
+      cd(this.selectedPageIndex, selectedPage);
+    });
   }
-  return undefined;
-}
 
-/**
- * hooks up select box with paging options
- * @param  {HTMLSelectElement} selectbox
- */
-export function initPagingSelectBox(selectbox: HTMLSelectElement) {
-  if (getPageCount() <= 1) {
-    selectbox.style.display = "none";
-    return;
+  /**
+   * Register subscriber callbacks to be called when the pageindex updates
+   * @param  {OnPagingCb} cb
+   * @returns number - index of the callback
+   */
+  public onPageUpdate(cb: OnPagingCb): number {
+    if (this.getPageCount() > 1) {
+      return this.onPageUpdateCbs.push(cb);
+    }
+    return undefined;
   }
-  waterfallDocsService.getDocs().pages.forEach((p, i) => {
-    let option = new Option(p.title, i.toString(), i === selectedPageIndex);
-    selectbox.add(option);
-  });
 
-  selectbox.style.display = "block";
-  selectbox.addEventListener("change", (evt) => {
-    let val = parseInt((evt.target as HTMLOptionElement).value, 10);
-    setSelectedPageIndex(val);
-  });
+  /**
+   * hooks up select box with paging options
+   * @param  {HTMLSelectElement} selectbox
+   */
+  public initPagingSelectBox(selectbox: HTMLSelectElement) {
+    const self = this;
+    if (this.getPageCount() <= 1) {
+      selectbox.style.display = "none";
+      return;
+    }
+    this.doc.pages.forEach((p, i) => {
```

Comment:
Just looked at it and it only seems to be executed once on init time. 

I am now wondering if someone has an options in the select box already could there be a use-case where they should not be cleared? I suppose a `please select...` placeholder or so should be cleared.

Anyway for now I'll clear them, since I can't think of any real-world case where they'd need to be kept.